commit 4bb0f4200ae67d203d59f19c7d38c01cc320ee03
Author: Artem Polyakov <artpol84@gmail.com>
Date:   Tue Apr 20 09:17:32 2021 -0700

    pmix/pmix2x: Fix the PMIx discovery logic
    
    See https://github.com/open-mpi/ompi/issues/8823 for the details.
    
    Signed-off-by: Artem Polyakov <artpol84@gmail.com>

diff --git opal/mca/pmix/pmix2x/pmix2x_component.c opal/mca/pmix/pmix2x/pmix2x_component.c
index 74a08a77058b..0771331a0679 100644
--- opal/mca/pmix/pmix2x/pmix2x_component.c
+++ opal/mca/pmix/pmix2x/pmix2x_component.c
@@ -118,9 +118,7 @@ static int external_component_query(mca_base_module_t **module, int *priority)
     char *t, *id;
 
     /* see if a PMIx server is present */
-    if (NULL != (t = getenv("PMIX_SERVER_URI")) ||
-        NULL != (t = getenv("PMIX_SERVER_URI2")) ||
-        NULL != (id = getenv("PMIX_ID"))) {
+    if (NULL != (t = getenv("PMIX_NAMESPACE"))) {
         /* if PMIx is present, then we are a client and need to use it */
         *priority = 100;
     } else {
